featuretools/tests/primitive_tests/test_install_primitives.py

```diff
@@ -0,0 +1,126 @@
+import os
+
+import pytest
+
+import featuretools
+from featuretools.primitives.base import PrimitiveBase
+from featuretools.primitives.install import (
+    get_installation_dir,
+    list_primitive_files,
+    load_primitive_from_file
+)
+
+try:
+    from builtins import reload
+except Exception:
+    from importlib import reload
+
+
+@pytest.fixture(scope='module')
+def primitives_to_install_dir():
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(this_dir, "primitives_to_install")
+
+
+@pytest.fixture(scope='module')
+def bad_primitives_files_dir():
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(this_dir, "bad_primitive_files")
+
+
+@pytest.fixture(scope='module')
+def primitives_to_install_archive():
+    # command to make this file: tar -zcvf primitives_to_install.tar.gz primitives_to_install/*.py
+    this_dir = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(this_dir, "primitives_to_install.tar.gz")
+
+
+def setup_installation():
+    installation_dir = get_installation_dir()
+    custom_max_file = os.path.join(installation_dir, "custom_max.py")
+    custom_mean_file = os.path.join(installation_dir, "custom_mean.py")
+    custom_sum_file = os.path.join(installation_dir, "custom_sum.py")
+
+    # make sure primitive files aren't there e.g from a failed run
+    for p in [custom_max_file, custom_mean_file, custom_sum_file]:
+        try:
+            os.unlink(p)
+        except Exception:
+            pass
+
+
+def cleanup_installation():
+    installation_dir = get_installation_dir()
+    files = list_primitive_files(installation_dir)
+    # then delete to clean up
+    for f in files:
+        os.unlink(f)
+
+
+def do_installation(install_path):
+    featuretools.primitives.install.install_primitives(install_path, prompt=False)
+
+    # must reload submodule for it to work
+    reload(featuretools.primitives.installed)
+    from featuretools.primitives.installed import CustomMax, CustomSum, CustomMean  # noqa: F401
+
+    installation_dir = get_installation_dir()
+    files = list_primitive_files(installation_dir)
+    custom_max_file = os.path.join(installation_dir, "custom_max.py")
+    custom_mean_file = os.path.join(installation_dir, "custom_mean.py")
+    custom_sum_file = os.path.join(installation_dir, "custom_sum.py")
+    assert set(files) == {custom_max_file, custom_mean_file, custom_sum_file}
+
+
+def test_install_primitives_directory(primitives_to_install_dir):
```

Comment:
We can probably combine these install primitives tests into a single parameterized test

Reply:
done. thanks for adding